### PR TITLE
implement (slow) seed expansion for small seeds

### PR DESCRIPTION
**Pull request by jneen**
where small is defined as < depth tokens.

### CODE REVIEW

File: src/main/scala/MarkovDict.scala [hunk 1]
```diff
@@ -80,8 +80,34 @@ class MarkovDict(val depth: Int,
     }
   }
 
+  private def expandSeed(seed: Array[String]) : Array[String] = {
```
Comment (stufflebear) [inline src/main/scala/MarkovDict.scala:83]: You could annotate this with the tail recursion flag `@tailrec`

File: src/main/scala/MarkovDict.scala [hunk 2]
```diff
@@ -80,8 +80,34 @@ class MarkovDict(val depth: Int,
...
+    }
+
+    val randIdx = randGen.nextInt(validInits.size)
+    validInits(randIdx)
+  }
+
+  private def improveSeed(seed: Array[String]) : Array[String] = {
+    if (seed.size < depth) return expandSeed(seed)
+
+    val tail = seed.takeRight(depth).toList
```
Comment (stufflebear) [inline src/main/scala/MarkovDict.scala:99]: Why the last d words?

File: src/main/scala/MarkovDict.scala (hunk 1 above)
Comment (jneen) [inline src/main/scala/MarkovDict.scala:83]: That was my intention - it wouldn't compile when I tried that though.  The depth of the recursion is bounded by `depth` though, so I thought it wasn't a huge deal.  Can you get it to work?

**Comment by stufflebear** [comment]
You can go ahead and merge this if you want to. I'll probably play around with some topic modeling for the seed.

**Comment by jneen** [comment]
Cool!

**Comment by jneen** [comment]
Also look at #15 before you do a lot of extra work (it changes the types of the keys).

**Pull request merged**